File: Data Structures & Algorithms/subsets/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    subsets(nums) { 
        const result = [] 
        this.rec(0, nums, [], result) 
        return result
    } 

    rec(i, nums, cur, res) {
        if (i === nums.length) {
            res.push(cur) 
            return
        } 


        let num = nums[i] 
        this.rec(i + 1, nums, [...cur, num], res) 
        this.rec(i + 1, nums, [...cur], res)
    }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-1.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        res = strs[0] 

        for s in strs: 
            i = 0 
            j = 0 

            while i < len(res) and j < len(s): 
                if res[i] == s[j]: 
                    i += 1 
                    j += 1 
                else:
                    break 

            res = res[0 : i] 

        return res